osu.Server.Spectator/Hubs/Metadata/MetadataHub.cs

```diff
@@ -65,9 +66,19 @@ public override async Task OnConnectedAsync()
                         versionHash = versionHash.Substring(versionHash.Length - 82, 32);
                 }
 
-                usage.Item = new MetadataClientState(Context.ConnectionId, Context.GetUserId(), versionHash);
+                usage.Item = new MetadataClientState(Context.ConnectionId, Context.GetUserId(), versionHash)
+                {
+                    UserStatus = UserStatus.Online
+                };
+
                 await broadcastUserPresenceUpdate(usage.Item.UserId, usage.Item.ToUserPresence());
                 await Clients.Caller.DailyChallengeUpdated(dailyChallengeUpdater.Current);
+
+                using (var db = databaseFactory.GetInstance())
+                {
+                    foreach (int friendId in await db.GetUserFriendsAsync(usage.Item.UserId))
+                        await Groups.AddToGroupAsync(Context.ConnectionId, FRIEND_PRESENCE_WATCHERS_GROUP(friendId));
```

Comment:
This doesn't send the initial "online" presence to the user in the case a friend is online. I think this is required.

Reply:
Makes sense, I only added this because I noticed a test failure. Have adjusted.

Reply:
This was an intentional choice, I haven't tested how stable does it. Basically you want to receive a giga notification on startup if 100 of your friends are online?

Reply:
I think I understand your concern now, have applied a fix in https://github.com/ppy/osu-server-spectator/pull/255/commits/fbdf5918ff46b56e8a2c742b9c83303c9e6d8f78.

Reply:
> Edit: Stable doesn't notify either.

It definitely does for me.. unless I'm hallucinating.

https://github.com/user-attachments/assets/558dfcdd-b190-4eda-9f39-c312038ab12c

Maybe you had this turned off in settings?